### PR TITLE
Make All Worlds Discoverable Proposal

### PR DESCRIPTION
**Pull request by contribuyente**
### Summary
In response to the governance proposal, [Make All Worlds Discoverable On Places dApp](https://governance.decentraland.org/proposal/?id=733a0260-587a-11ee-a5cc-3955d256c410), this pull request not only enhances the discoverability of "Worlds" in the Decentraland Places dApp but also undertakes a significant cleanup of redundant database and code infrastructure. All Worlds, irrespective of their association with LAND owners or renters, are now discoverable.

### Changes
- Eliminated the hidden column from the database.
- Purged the `AND hidden=false` condition from the findWorld and countWorlds queries.
- Removed all occurrences and usages of the hidden property across various files.
- Deleted the user-facing disclaimer in the UI regarding the discoverability constraints of Worlds.
- Associated tasks that relied on the hidden attribute have been appropriately removed.

### Impact
This updated change is a leap towards the community's vision of a more inclusive and engaging Decentraland. All Worlds are now on an even playing field, fostering a more democratic content discovery process.

### Reviewers
A call for Foundation devs and proactive community members to ensure these changes meet the intent and spirit of the governance proposal. Feedback is immensely valuable to guarantee a seamless and impactful transition.

### CODE REVIEW

**Comment by contribuyente** [comment]
Following the positive outcome of the [binding governance proposal](https://governance.decentraland.org/proposal/?id=733a0260-587a-11ee-a5cc-3955d256c410), which garnered a significant 10M VP in favor, I'm reaching out to leading contributors of this repository, @braianj and @2fd, for a review of this pull request. If any aspects of this implementation are found lacking or misaligned, please don't hesitate to close it and take any necessary actions. Appreciate your attention and collaboration. Thank you.

**Comment by 2fd** [comment]
Hi @contribuyente, thanks for contributing to the project!

The `hidden` feature was developed only because of the restriction we had to publish all worlds, so I will ask you to remove it entirely from the project, including the database, and to modify the database indices to prevent a regression in the performance of the queries.

To do so, you will need to:
- run the command `npm run migrate create remove-hidden-prop-from-places`
- using the [migration of the hidden prop](https://github.com/decentraland/places/blob/master/src/migrations/1686681381551_add-places-hidden-property.ts) as guide:
  - drop the `hidden` column using the `pgm.dropColumn` function
  - drop all indices that include `hidden` with `pgm.dropIndex`
  - re-create the exact same indices without the `hidded` with `pgm.createIndex`
  - don't forget to define the `down` function accordingly
- remove all usage of the `hidden` prop from the following files:
  - [src/__data__/placeRoad.ts](../tree/master/src/__data__/placeRoad.ts#L34)
  - [src/entities/Slack/utils.ts](../tree/master/src/entities/Slack/utils.ts#L32)
  - [src/entities/Place/types.ts](../tree/master/src/entities/Place/types.ts#L35)
  - [src/entities/World/schemas.ts](../tree/master/src/entities/World/schemas.ts#L179-L182)
  - [src/entities/Place/model.ts](../tree/master/src/entities/Place/model.ts#L410) (the entire `updateIndexWorlds` method)
  - [src/entities/CheckScenes/task/processContentEntityScene.ts](../tree/master/src/entities/CheckScenes/task/processContentEntityScene.ts#L115)
  - [src/entities/Place/schemas.ts](../tree/master/src/entities/Place/schemas.ts#L237-L239)
  - [src/__data__/placeGenesisPlazaWithAggregatedAttributes.ts](../tree/master/src/__data__/placeGenesisPlazaWithAggregatedAttributes.ts#L420)
  - [src/__data__/placeGenesisPlaza.ts](../tree/master/src/__data__/placeGenesisPlaza.ts#L416)
  - [src/entities/CheckScenes/task/taskRunnerSqs.ts](../tree/master/src/entities/CheckScenes/task/taskRunnerSqs.ts#L40)
  - [src/entities/CheckScenes/task/taskRunnerSqs.ts](../tree/master/src/entities/CheckScenes/task/taskRunnerSqs.ts#L76)
- at last, you just need to remove the task that checks for world index:
   - [src/entities/World/task/processWorldsIndexing.ts.ts](../tree/master/src/entities/World/task/processWorldsIndexing.ts#L18)
   - [src/entities/Place/model.test.ts](../tree/master/src/entities/Place/model.test.ts#L813)
   - [src/server.ts](../tree/master/src/server.ts#L51)

**Comment by contribuyente** [comment]
Hi @2fd,

Thank you for the comprehensive guidance on this matter. It's clear to see why you're a core contributor, given the depth of insight you've provided here.

I understand the rationale behind the original hidden feature and the steps required to phase it out. I'll ensure that all related components are removed appropriately from the database and relevant files, and I'll attend to the indices to maintain the performance integrity of the queries.

Your clear outline makes this task straightforward, and I appreciate you taking the time to list out each and every file that needs attention.

Once I've made these changes, I'll push the updated pull request for further review. If there are any additional concerns or modifications required, don't hesitate to let me know.

**Comment by contribuyente** [comment]
Hey @2fd,

I've just pushed all the changes mentioned above. Would you please give it a look when you have a moment? Thanks 🙏 .

**Comment by contribuyente** [comment]
Hello @braianj, thank you for the feedback. I did as you said and moved the `dropColumn` call, also linted and setup the pre-commit hooks. I noticed some tests were failing too and I fixed them, but there are 3 test cases that fail and I don't understand the error, it doesn't seem to be related to the `hidden` column per se:

Screenshot 2023-10-17 at 16 47 14

They are all related to this "fetcher" not being set.

**Comment by braianj** [comment]
@contribuyente It could be your node version. Please notice that the min supported version is 18.18.X. Try with that.

**Comment by contribuyente** [comment]
Hi @braianj it was indeed the node version, thank you. It seems [all checks are passing](https://github.com/decentraland/places/actions/runs/6565018135/job/17832606109) now.

Screenshot 2023-10-19 at 10 48 21

File: src/migrations/1697479763911_make-all-words-discoverable.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  pgm.dropIndex(PlaceModel.tableName, ["hidden", "updated_at"])
+  pgm.dropIndex(PlaceModel.tableName, ["hidden", "like_rate"])
+
+  pgm.dropColumn(PlaceModel.tableName, shorthands)
+
+  pgm.createIndex(PlaceModel.tableName, ["world_name"])
+  pgm.createIndex(PlaceModel.tableName, ["base_position"])
+  pgm.createIndex(PlaceModel.tableName, ["updated_at"])
+  pgm.createIndex(PlaceModel.tableName, ["like_rate"])
+}
```
Comment (braianj) [inline src/migrations/1697479763911_make-all-words-discoverable.ts:26]: Here you'll need to drop the same index created before and then create the new index with the same where but without the hidden, in order to make a better perform.

```suggestion
export async function up(pgm: MigrationBuilder): Promise<void> {
  pgm.dropIndex(PlaceModel.tableName, ["hidden", "world_name"])
  pgm.dropIndex(PlaceModel.tableName, ["hidden", "base_position"])
  pgm.dropIndex(PlaceModel.tableName, ["updated_at"])
  pgm.dropIndex(PlaceModel.tableName, ["like_rate"])

  pgm.dropColumn(PlaceModel.tableName, shorthands)

  pgm.createIndex(PlaceModel.tableName, ["world_name"], {
    where: "disabled is false and world is true",
  })
  pgm.createIndex(PlaceModel.tableName, ["base_position"], {
    where: "disabled is false and world is false",
  })
  pgm.createIndex(PlaceModel.tableName, ["updated_at"], {
    where: "disabled is false and world is false",
  })
  pgm.createIndex(PlaceModel.tableName, ["like_rate"], {
    where: "disabled is false and world is false",
  })
}
```

**Review by braianj — CHANGES_REQUESTED** [review]
Please make those changes and we'll be go to do. Thanks!

**Comment by contribuyente** [comment]
Your suggestion has been committed @braianj 🤝

File: src/migrations/1697479763911_make-all-words-discoverable.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    default: false,
+    notNull: true,
+  },
+}
+
+export async function up(pgm: MigrationBuilder): Promise<void> {
+  pgm.dropIndex(PlaceModel.tableName, ["hidden", "world_name"])
+  pgm.dropIndex(PlaceModel.tableName, ["hidden", "base_position"])
+  pgm.dropIndex(PlaceModel.tableName, ["updated_at"])
+  pgm.dropIndex(PlaceModel.tableName, ["like_rate"])
```
Comment (contribuyente) [inline src/migrations/1697479763911_make-all-words-discoverable.ts:18]: Is it okay that these two don't have the `"hidden"` column? They were removed in your suggested commit @braianj

Comment (braianj) [reply src/migrations/1697479763911_make-all-words-discoverable.ts:18]: Yes if you look when the index has been created, those don't have hidden column either, the hidden is inside the where 

```
pgm.createIndex(PlaceModel.tableName, ["updated_at"], {
    where: "disabled is false and hidden is false and world is false",
  })
  pgm.createIndex(PlaceModel.tableName, ["like_rate"], {
    where: "disabled is false and hidden is false and world is false",
  })

```

**Review by braianj — APPROVED** [review]

**Review by braianj — APPROVED** [review]
Great job. 🚀 
thank you for contributing to the project. 👍

**Comment by contribuyente** [comment]
Thank you both @2fd and @braianj 🤝 🚀

**Pull request merged**